File: Go/873.length-of-longest-fibonacci-subsequence.go
// 方法1： 暴力判断
func lenLongestFibSubseq(A []int) int {
    total := len(A)
    m := make(map[int]int, total)
    for _, val := range A {
        m[val] = 1
    }
    max := 0
    for i := 0; i < total; i ++ {
        for j := i+1; j < total; j ++ {
            a, b := A[j], A[i] + A[j]
            count := 2
            flag := true
            for flag {
               if _, ok := m[b]; ok {
                    a, b = b, a + b
                    count ++
               } else {
                   flag = false
               }
            }
            if count >= 3 && count > max {
                max = count
            }
        } 
    }
    return max
}

File: PHP/873.length-of-longest-fibonacci-subsequence.php
<?php

class Solution {

    /**
     * @param Integer[] $A
     * @return Integer
     */
    function lenLongestFibSubseq($A) {
        $m = [];
        $keys = array_fill_keys($A, 1);
        $total = count($A);
        for ($i=0; $i<$total; $i++) {
            for ($j=$i+1; $j<$total; $j++) {
                $a = $A[$j];
                $b = $A[$i] + $A[$j];
                $count = 2;
                while (isset($keys[$b])) {
                    $t = $a;
                    $a = $b;
                    $b = $t + $b;
                    $count ++;
                }
                if ($count > 2) {
                    $m[] = $count;
                }
            }
        }
        return $m ? max($m) : 0;
    }
}
